Automattic/newspack-plugin
# Feat/use newspack scripts

### All Submissions:






### Changes proposed in this Pull Request:



Using [`newspack-scripts`](https://github.com/Automattic/newspack-scripts) to abstract away some common tasks & dependencies. 

Some remarks:

- `rtlcss` options are not provided in `newspack-scripts`, because they do not differ from the defaults
- The two `@babel/*` packages are added only on account of `happychat-client`. It would not build without these.
- Interestingly, `eslint` has a [years-old issue](https://github.com/eslint/eslint/issues/3458) with resolving dependencies of extended configurations. Luckily there's [a solution in a form of a patch](https://www.npmjs.com/package/@rushstack/eslint-patch).
- This PR moves the npm dependencies to proper slots. `@wordpress/*` packages are now listed in `devDependencies`, since these are not used in the built JS – `calypso-build` uses dependence extraction to use the versions of there packages provided by WordPress environment. It's still advisable to list these dependencies so that these packages can be used in test environment. 

#### Caveats

These changes may break some editor integrations. My case was using `prettier-atom` in Atom editor – I had to disable it's use of `prettier-eslint` since it did not resolve `eslint` dependencies well.

### How to test the changes in this Pull Request:

1. Run `npm run build`
2. Do a short smoke test – do the wizards load all right?
2. Observe the built assets are **minified**
1. Verify [`postcss-focus-within`](https://www.npmjs.com/package/postcss-focus-within) is used 
    1. Run `cat dist/common.css | grep '\[focus-within\]'` – observe that there is output
2. Verify [`autoprefixer`](https://github.com/postcss/autoprefixer) is used:
    1. Add `.niceclassname {user-select: none;}` style to `assets/components/src/style.scss`
    2. Run `npm run build`
    3. Run `cat dist/commons.css | grep 'niceclassname'` – observe there's a `-webkit-user-select` rule used 
3. Verify `eslint` can be run (`npm run lint:js`) – there should be no issues 
4. Run `npm run start`, see that it runs webpack in watch mode, as before
5. Verify `stylelint` can be run (`npm run lint:scss`) – there should be no issues
6. Verify `commitizen` works by running `npm run cm` – a prompt for a structured commit message should appear, just line on `master`
7.  Verify the `commit-msg` hook does lint the commit message:
    1. reinstall the hook – `composer install`
    2. try to commit with an unstructured commit message (e.g. `git commit -m "hehe"`) and observe it failing
8. Change to the release process can effectively be tested during a release, but a smoke test can be performed by comparing that the output of `$ npm run semantic-release -- --no-ci --dry-run` on `master` and on this branch is the same

### Other information:
